Modules/Core/Common/include/itkCommonEnums.h

```diff
@@ -0,0 +1,286 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+// Place common enumerations to be used throughout the toolkit in this file
+#ifndef itkCommonEnums_h
+#define itkCommonEnums_h
+
+#include "itkIntTypes.h"
+#include <ostream>
+
+namespace itk
+{
+
+/** \class CommonEnums
+ *
+ * \brief Common enums used across the toolkit
+ *
+ * This class encapsulates the common enum classes. It is required for
+ * wrapping.
+ *
+ * \ingroup ITKCommon
+ **/
+class CommonEnums
+{
+public:
+  // Used in Input/Output for Images/Mesh/Transform types
+  /**
+   * \class IOPixel
+   * \ingroup ITKCommon
+   * Enums used to manipulate the point/cell pixel type. The pixel type provides
+   * context for automatic data conversions (for instance, RGB to
+   * SCALAR, VECTOR to SCALAR).
+   */
+  enum class IOPixel : uint8_t
+  {
+    UNKNOWNPIXELTYPE,
+    SCALAR,
+    RGB,
+    RGBA,
+    OFFSET,
+    VECTOR,
+    POINT,
+    COVARIANTVECTOR,
+    SYMMETRICSECONDRANKTENSOR,
+    DIFFUSIONTENSOR3D,
+    COMPLEX,
+    FIXEDARRAY,
+    ARRAY,
+    MATRIX,
+    VARIABLELENGTHVECTOR,
+    VARIABLESIZEMATRIX
+  };
+
+  /**
+   * \class IOComponent
+   * \ingroup ITKCommon
+   * Enums used to manipulate the component type. The component type
+   * refers to the actual storage class associated with either a
+   * SCALAR pixel type or elements of a compound pixel.
+   */
+  enum class IOComponent : uint8_t
+  {
+    UNKNOWNCOMPONENTTYPE,
+    UCHAR,
+    CHAR,
+    USHORT,
+    SHORT,
+    UINT,
+    INT,
+    ULONG,
+    LONG,
+    LONGLONG,
+    ULONGLONG,
+    FLOAT,
+    DOUBLE,
+    LDOUBLE
+  };
+
+  /**
+   * \class IOFile
+   * \ingroup ITKCommon
+   * Enums used to specify write style: whether binary or ASCII. Some
+   * subclasses use this, some ignore it.
+   */
+  enum class IOFile : uint8_t
+  {
+    ASCII = 0,
+    Binary = 1,
+    TypeNotApplicable = 2,
+    // for backward compatibility
+    BINARY = 1,           // Spelling difference between IOMeshBase and IOImageBase
+    TYPENOTAPPLICABLE = 2 // Spelling difference between IOMeshBase and IOImageBase
+  };
+
+  // TODO: Enable for backward compatibility using File Enum = IOFileEnum; // For backward compatibility
```

Comment:
There are a bunch of these TODO comments in this file which should be addressed.